File: python/python/day4_functions.py
# Day 4 - Python Functions

# Factorial function
def factorial(n):
    result = 1
    for i in range(1, n+1):
        result *= i
    return result

print("Factorial of 5 =", factorial(5))

# Palindrome check function
def is_palindrome(s):
    return s == s[::-1]

word = "level"
print(f"Is '{word}' a palindrome? {is_palindrome(word)}")

File: javascript/javascript/day4_functions.js
// Day 4 - JavaScript Functions

// Factorial function
function factorial(n) {
    let result = 1;
    for (let i = 1; i <= n; i++) {
        result *= i;
    }
    return result;
}
console.log("Factorial of 5 =", factorial(5));

// Palindrome check function
function isPalindrome(str) {
    return str === str.split('').reverse().join('');
}
let word = "level";
console.log(`Is '${word}' a palindrome? ${isPalindrome(word)}`);
